webapp/src/main/java/com/box/l10n/mojito/service/thirdparty/ThirdPartyTMSSmartling.java

```diff
@@ -91,11 +136,104 @@ public void createImageToTextUnitMappings(String projectId, List<ThirdPartyImage
+            AndroidStringDocumentWriter writer = new AndroidStringDocumentWriter(mapper.readFromSourceTextUnits(result));
+            file.setFileContent(writer.toText());
+
+        } catch (ParserConfigurationException | TransformerException e) {
+            logger.error("An error ocurred when processing a push batch", e);
+            throw new RuntimeException(e);
+        }
+
+        if (!options.isDryRun()) {
+            smartlingClient.uploadFile(projectId, file.getFileName(), "android",
+                    file.getFileContent(), options.getPlaceholderFormat(), options.getCustomPlaceholderFormat());
+        }
+
+        return file;
+    }
+
+    private Stream<List<TextUnitDTO>> partitionSingulars(Long repositoryId,
+                                                         List<String> localesTags,
+                                                         String skipTextUnitsWithPattern,
+                                                         String skipAssetsWithPathPattern) {
+        return StreamSupport.stream(Iterables.partition(textUnitSearcher.search(baseParams(repositoryId, localesTags, skipTextUnitsWithPattern,
+                skipAssetsWithPathPattern, true, true, null)), batchSize).spliterator(), false);
+    }
+
+    private Stream<List<TextUnitDTO>> partitionPlurals(Long repositoryId,
+                                                       List<String> localesTags,
+                                                       String skipTextUnitsWithPattern,
+                                                       String skipAssetsWithPathPattern) {
+        return StreamSupport.stream(Iterables.partition(textUnitSearcher.search(baseParams(repositoryId, localesTags, skipTextUnitsWithPattern,
+                skipAssetsWithPathPattern, false, false, "%")), batchSize).spliterator(), false);
     }
+
+    private TextUnitSearcherParameters baseParams(Long repositoryId,
+                                                  List<String> localesTags,
+                                                  String skipTextUnitsWithPattern,
+                                                  String skipAssetsWithPathPattern,
+                                                  boolean pluralFormsFiltered,
+                                                  boolean pluralFormsExcluded,
+                                                  String pluralFormOther) {
+        TextUnitSearcherParameters result = new TextUnitSearcherParameters();
+        result.setRepositoryIds(repositoryId);
+        result.setLocaleTags(localesTags);
+        result.setRootLocaleExcluded(false);
+        result.setDoNotTranslateFilter(false);
+        result.setSearchType(SearchType.ILIKE);
+        result.setStatusFilter(StatusFilter.TRANSLATED);
+        result.setPluralFormsFiltered(pluralFormsFiltered);
+        result.setPluralFormsExcluded(pluralFormsExcluded);
+        result.setSkipTextUnitWithPattern(skipTextUnitsWithPattern);
```

Comment:
just realizing that the text unit searcher could have been kept more generic by adding a search type: "not like" and search on name or path.

if so, could you clean that up in a following PR, ideally we should try to not have to much boilerplate in the searcher, it has already grown quite a bit.